# HG changeset patch
# User Ben Abbott  <bpabbott@mac.com>
# Date 1291336033 18000
#      Thu Dec 02 19:27:13 2010 -0500
# Node ID 9a08fa5b33bd9fa083050a43c9468bfaf330ec8a
# Parent  6ab93f01df07577e1638dec7616203ee6d62b636
image.m: Only flip y-axis if nextplot property equal "replace".

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,10 +1,11 @@
 2010-12-02  Ben Abbott <bpabbott@mac.com>
 
+	* image/image.m: Only flip y-axis if nextplot property equal "replace".
 	* plot/contour3.m: Modify demo to avoid clipping mesh.
 	* plot/fill.m: Allow multiple filled polygons.
 	* plot/subplot.m: Add suppport for "align" and "replace" options.
 
 2010-11-26  John W. Eaton  <jwe@octave.org>
 
 	* pkg/pkg.m: Append directories to EXEC_PATH instead of
 	prepending them.  Use pathsep instead of ":".
diff --git a/scripts/image/image.m b/scripts/image/image.m
--- a/scripts/image/image.m
+++ b/scripts/image/image.m
@@ -148,18 +148,20 @@ function h = __img__ (x, y, img, varargi
       mn = intmin (c);
       mx = intmax (c);
       set (ca, "clim", double ([mn, mx]));
     endif
   endif
 
   set (ca, "view", [0, 90]);
 
-  # Always reverse y-axis for images, even on existing plots
-  set (ca, "ydir", "reverse");
+  if (strcmp (get (ca, "nextplot"), "replace"))
+    # Always reverse y-axis for images, unless hold is on
+    set (ca, "ydir", "reverse");
+  endif
 
   if (nargout > 0)
     h = tmp;
   endif
 
 endfunction
 
 %!demo
